reposerver/repository/repository.go

```diff
@@ -81,17 +81,17 @@ func (s *Service) ListApps(ctx context.Context, q *apiclient.ListAppsRequest) (*
 	if err != nil {
 		return nil, err
 	}
-	if apps, err := s.cache.ListApps(q.Repo.Repo, q.Revision); err == nil {
-		log.Infof("cache hit: %s/%s", q.Repo.Repo, q.Revision)
-		return &apiclient.AppList{Apps: apps}, nil
-	}
 	apps, resolvedRevision, err := r.ListApps(q.Revision)
```

Comment:
In this case, we are not taking any advantage of caching. We need to resolve revision before locking the repository. Check the cache and then lock repo, fetch, checkout, search for apps etc.

Same is in `GetRevisionMetadata`

Reply:
I think we need to lock the repo on any operation. Let me have a deep think

Reply:
We use `go-git` to execute `ls-remote` in memory. This allows to check if operation result is available in cache without locking the repository.

Reply:
I did not explicitly mention another issue here. Looks like we execute `ListApps` before checking if result is available in cache or not. That is what I mean by `we are not taking any advantage of caching`

Reply:
I've reverted the change, as the fix is more involved. I'll create another issuep